### ISSUE TITLE
Weird error messages from failed migrations

### ISSUE THREAD

**Issue by rzane** [livehelpnow/tds_ecto#59]
When a migration fails, `Tds.Ecto` generates cryptic error messages:

```
** (Tds.Error) Line 1 (Error 3971): The server failed to resume the transaction. Desc:3400000008.
```

If I add the [`@disable_ddl_transaction true`](https://hexdocs.pm/ecto/Ecto.Migration.html#module-transactions) attribute, I am able to read the real error message.

This isn't a huge deal, but it took me a while to figure out why my migration was failing. At least by creating an issue, other people might be able to figure out what's wrong with their migration.

**Comment by mjaric** [comment]
if possible, could you please provide us the case which fails with this message?

**Comment by rzane** [comment]
I've created an app here: https://github.com/rzane/tds_ecto_repro.

Here's the migration with the wierd error message: https://github.com/rzane/tds_ecto_repro/blob/master/priv/repo/migrations/20180208141326_one.exs

To reproduce, clone the repo and run `mix do deps.get, ecto.create, ecto.migrate`. You might need to change the connection config in `config/dev.exs`.

**Comment by mjaric** [comment]
Thanks! I will check it.

**Comment by indyone** [comment]
@mjaric I faced a similar case, where I got exactly the same error,

```
** (Tds.Error) Line 1 (Error 3971): The server failed to resume the transaction. Desc:3300000001
```

In my case, i was passing an `Ecto.Multi` to a `Repo.transaction/2` and one of the queries was failing, making the transaction to rollback. If the rollback, also fails then the above error is the one that gets raised, silently ignoring the actual error of my failed query.

To find out what was the actual error... I "hacked" the `Tds.Protocol` module so that it outputs in the console all errors.

I think that somehow, we should include in the `Tds.Error` the *inner* or *related* errors, for easier error re-solving... What do you think?

**Comment by mjaric** [comment]
It is probably due bad return flow in tds app, could be that some func. raises error instead of returning labeled value. 

What version of TDS are you using?

**Comment by mjaric** [comment]
Solved in tds_ecto 2.2
